src/privileged/vpnRecommender/api.js

```diff
@@ -0,0 +1,537 @@
+const VPN_LANDING_PAGE_DEFAULT_PARAMS = {
+  "utm_source": "firefox-browser",
+  "utm_medium": "firefox-browser",
+  "utm_campaign": "shield_vpn",
+};
+
+const log = function(...args) {
+  if (!Preferences.get(DEBUG_MODE_PREF)) return;
+  console.log(...args);
+};
+
+const DOORHANGER_MESSAGES = {
+  "captive-portal": {
+    header: "It appears you are browsing on an unsecured wireless network.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connect, no matter where you are.",
+  },
+  "privacy-hostname": {
+    header: "Make Firefox even more secure with ProtonVPN.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connect, no matter where you are.",
+  },
+  "streaming-hostname": {
+    header: "Make Firefox even more secure with ProtonVPN.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connect, no matter where you are.",
+  },
+  "catch-all": {
+    header: "Make Firefox even more secure with ProtonVPN.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connect, no matter where you are.",
+  },
+};
+
+this.vpnRecommender = class extends ExtensionAPI {
+
+  getAPI(context) {
+    const that = this;
+
+    console.log("context", context);
+
+    this.jsms = {}; // workaround for not being able to use jsms with moz:// urls
+
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/Doorhanger.jsm"), this.jsms);
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/VpnRelatedHostnames.jsm"), this.jsms);
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/RecentWindow.jsm"), this.jsms);
+
+    this.Doorhanger = this.jsms.Doorhanger;
+    this.VpnRelatedHostnames = this.jsms.VpnRelatedHostnames;
+    this.RecentWindow = this.jsms.RecentWindow;
+
+    this.cleanUpFunctions = [];
+
+    this.addCleanUpFunction(() => {
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/Doorhanger.jsm"));
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/VpnRelatedHostnames.jsm"));
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/RecentWindow.jsm"));
+
+      that.killNotification();
+    });
+
+    this.extensionUrl = context.extension.getURL();
+
+    const EventManager = ExtensionCommon.EventManager;
+
+    this.listenForAddonDisableOrUninstall(context.extension.id); // https://github.com/mozilla/shield-studies-addon-utils/issues/247
+
+    AddonManager.addAddonListener(this);
+    return {
+      experiments: {
+        vpnRecommender: {
+          echo(str) {
+            log(`echoing: ${str}`);
+          },
+
+          start(variation, isFirstRun) {
+            that.start(variation, isFirstRun);
+          },
+
+          getInternals() {
```

Comment:
What is this?

Reply:
Old artifact left from early testing. Removed it.